service/service_test.go

```diff
@@ -1,15 +1,38 @@
 package service
 
 import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/giantswarm/micrologger/microloggertest"
 	"github.com/spf13/viper"
+	v1 "k8s.io/api/core/v1"
 
 	"github.com/giantswarm/app-operator/flag"
 )
 
 func Test_Service_New(t *testing.T) {
+	// fake server to initialize helm client
+	// there are two calls to this server during initialization,
+	// getting the name of tiller pod and port forwarding to it
+	h := func(w http.ResponseWriter, r *http.Request) {
+		podList := v1.PodList{
+			Items: []v1.Pod{},
+		}
+		pods, err := json.Marshal(podList)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(pods)
+	}
+	ts := httptest.NewServer(http.HandlerFunc(h))
+	defer ts.Close()
```

Comment:
This part I don't get. Is the `k8sclient` lib setting up helm client as well?

Reply:
Sorry this was a bit rough. With the k8sclient collection the service test errors if a JSON response is not returned. This was copy pasta from chart-operator.

We know these service tests UTs can get complex. I'll update the comments but maybe it should be dropped like we did in other projects?

Reply:
Tidied up the UT a bit and fixed the comments.

Reply:
Do you know what produces the request?

Reply:
BTW it's not a blocker. Merge if you need. We con continue the discussion.

Reply:
@kopiczko Sorry was getting the rest of my PRs sorted. Its super weird it seems to be doing a DNS lookup because of https://github.com/giantswarm/k8sclient/blob/059a94fd04e59aceab384413cf9608ae1640a2ca/clients.go#L115.

I simplified to just return empty JSON as that is the minimum it will accept.

```
{"caller":"github.com/giantswarm/app-operator/vendor/github.com/giantswarm/operatorkit/client/k8srestconfig/k8s_rest_config.go:158","level":"debug","message":"creating out-cluster REST config","time":"2019-12-05T15:15:36.532146+00:00"}
{"caller":"github.com/giantswarm/app-operator/vendor/github.com/giantswarm/operatorkit/client/k8srestconfig/k8s_rest_config.go:173","level":"debug","message":"created out-cluster REST config","time":"2019-12-05T15:15:36.532243+00:00"}
--- FAIL: Test_Service_New (0.04s)
    --- FAIL: Test_Service_New/case_0:_valid_config_returns_no_error (0.04s)
        /Users/rossf7/.gvm/pkgsets/go1.13.1/global/src/github.com/giantswarm/app-operator/service/service_test.go:77: error == [{/Users/rossf7/.gvm/pkgsets/go1.13.1/global/src/github.com/giantswarm/app-operator/service/service.go:91: } {/Users/rossf7/.gvm/pkgsets/go1.13.1/global/src/github.com/giantswarm/app-operator/vendor/github.com/giantswarm/k8sclient/clients.go:117: } {Get http://kubernetes/api?timeout=10s: dial tcp: lookup kubernetes: no such host}], want nil
```

Going to go with this but maybe we can fix it in k8sclient somehow?